### ISSUE TITLE
Handle components project as another content Git repo

### ISSUE THREAD

**Issue by Swiip** [aya-experience/citation#46]
By "components project", I mean, the user project where he develop his components.

After some brainstorming about the usage and the initialization of citation, I consider that having the server as a dependency of the component is more weird than anything else.

I tried to find a simpler model and I came up with that:
- citation-server is a standalone server usable through cli or a simple JS api.
- the config point 2 repos: one for the data, another for the "components project"
- the server check periodically both repo and update the website accordingly
- it implies that the server has to be able to manage (build and compile) the components project

To have this, we will have to remove the demo "components project" from the core citation repo as the lifecycle of the projects will be different. But to be honest having an example inside the main repo was not really logic anyway.

**Comment by Slashgear** [comment]
Furthermore we could handle Github/Bitbucket/Gitlab hooks to update citation website in order to avoid checking periodically.

This deported model for component is very cool, we could imagine community catalog for sharing citation components.

**Comment by Swiip** [comment]
Yep for GitHub and co. But we have to keep pure Git as users will want to be able to keep data in their own closed infra.

Sharing components is a great idea. It would implies to manage multiple components repo but why not! All the components will have to be of the same framework or it could become complicated ;)

**Comment by Swiip** [comment]
My concept of "components project" and your idea of "community components" has a lots of implications.

It would implies that the JS project from the user would not be a website by itself but only a component library.

To be honest, my demo project already looks a lot like this. It could be an evolution I'm willing to make but it's pretty important.

The real bundling of the project would be transfered from the user project to Citation, it's an important responsibility but it could be simpler to manage a centralized packaging.

**Comment by Swiip** [comment]
Still working on this idea and all the implications it brings.

Components projects could become pure generic components library, it would be great for interoperability. It could also be great for simplicity of creating components projects.

But it brings new important questions:
- How components will handle their logic? Each one with it's own framework? We produce a common API like a Redux?
- Where to put CSS? Each components can brings its style but what about global CSS for a website?
- Citation will have to come with it's own bundling solution.
- Citation will have to provide a dev mode with live reload as the components projects will not be "standalone webapps"

Finally, it means that user lose the ability of choosing its tooling. For production build, it could make sense to give Citation ability to package a fully optimized website but for development process, it's really less simple.

**Comment by Swiip** [comment]
I think I got something.

I created a `citation-react-builder` project which basically is an empty create-react-app project. It's only code is in the index.js which load the Citation router.

`citation-server` use this builder project to bundle the website using all the power of create-react-app without having to manage (and maintain) the build itself.

Better, we could easily imagine that a power user will be able to replace by configuration the default builder project with its own and user whatever tooling he wants.

At the end we could consider that we splitted the demo project in 3:
- The server deps and the configration
- The builder project which handle (only) the packaging
- The components project which is a simple JS library

Like this and with a bit of configuration magic. Everything can work by default and still be extensible.
